runway/hooks/staticsite/auth_at_edge/templates/check_auth/__init__.py

```diff
@@ -0,0 +1,249 @@
+    querystring = request.get('querystring')
+    # pylint: disable=line-too-long
+    request_query_string = ("?%s" % querystring) if querystring else ""
+    requested_uri = '%s%s' % (request['uri'], request_query_string)
+    nonce = generate_nonce()
+
+    try:
+        # Extract the cookies received from Cognito
+        extracted = extract_and_parse_cookies(
+            request['headers'],
+            config['client_id']
+        )
+        token_user_name = extracted.get('tokenUserName')
+        id_token = extracted.get('idToken')
+        refresh_token = extracted.get('refreshToken')
+
+        # If the token user name or id token are missing then we need
+        # new credentials
+        if not token_user_name or not id_token:
+            msg = "No valid credentials present in cookies"
+            LOGGER.error(msg)
+            raise Exception(msg)
+
+        # Get the expiration date from the id token.
+        decoded_token = decode_token(id_token)
+        expiration = decoded_token.get('exp')
+        exp_date = datetime.datetime.fromtimestamp(expiration)
+        now = datetime.datetime.now()
+
+        # If we have a refresh token and the expiration date has passed
+        # then forward the user to the refresh agent
+        if now > exp_date and refresh_token:
+            headers = {
+                # Redirect the user to the refresh agent
+                'location': [
+                    {
+                        'key': 'location',
+                        'value': 'https://%s%s?%s' % (
+                            domain_name,
+                            config.get('redirect_path_auth_refresh'),
+                            urlencode({
+                                'requestedUri': requested_uri,
+                                'nonce': nonce
+                            })
+                        )
+                    }
+                ],
+                'set-cookie': [
+                    # Set the Nonce Cookie
+                    {
+                        'key': 'set-cookie',
+                        'value': 'spa-auth-edge-nonce=%s; %s' % (
+                            quote_plus(nonce),
+                            config.get('cookie_settings').get('nonce')
+                        )
+                    }
+                ]
+            }
+            # Add all CloudFrontHeaders
+            headers.update(config.get('cloud_front_headers'))
+            return {
+                'status': '307',
+                'statusDescription': 'Temporary Redirect',
+                'headers': headers
+            }
+
+        # Validate the token information against the Cognito JWKS
+        validate_jwt(
+            id_token,
+            config.get('token_jwks_uri'),
+            config.get('token_issuer'),
+            config.get('client_id'))
+
+        return request
+    # We need new authorization. Get the user over to Cognito
+    # pylint: disable=broad-except
+    except Exception as _: # noqa
```

Comment:
if you arn't going to inspect the exception, you can just do `except Exception:`